newIDE/app/src/EventsSheet/InstructionEditor/TextEditorDialog.js

```diff
@@ -0,0 +1,301 @@
+      gd.asGroupEvent(event).setBackgroundColor(
+        parseInt(backgroundColors.red),
+        parseInt(backgroundColors.green),
+        parseInt(backgroundColors.blue)
+      );
+    }
+  };
+
+  getComment = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      return gd.asCommentEvent(event).getComment();
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      return gd.asGroupEvent(event).getName();
+    } else {
+      console.error(
+        'Dialog was opened for an unsupported event type: ' + eventType
+      );
+      return '';
+    }
+  };
+
+  getTextColors = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    let textColors;
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      const commentEvent = gd.asCommentEvent(event);
+      textColors = {
+        red: commentEvent.getTextColorRed(),
+        green: commentEvent.getTextColorGreen(),
+        blue: commentEvent.getTextColorBlue(),
+      };
+      return textColors;
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      var groupEvent = gd.asGroupEvent(event);
+      const r = groupEvent.getBackgroundColorR(),
+        g = groupEvent.getBackgroundColorG(),
+        b = groupEvent.getBackgroundColorB();
+
+      textColors = (r + g + b) / 3 > 200 ? 'black' : 'white'; //Because text color is not supported by Core
+      return textColors;
+    } else {
+      console.error('Dialog was opened for an unsupported event text color.');
+      return '';
+    }
+  };
+
+  getBackgroundColors = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    let backgroundColors;
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      const commentEvent = gd.asCommentEvent(event);
+      backgroundColors = {
+        red: commentEvent.getBackgroundColorRed(),
+        green: commentEvent.getBackgroundColorGreen(),
+        blue: commentEvent.getBackgroundColorBlue(),
+      };
+
+      return backgroundColors;
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      const groupEvent = gd.asGroupEvent(event);
+      backgroundColors = {
+        red: groupEvent.getBackgroundColorR(),
+        green: groupEvent.getBackgroundColorG(),
+        blue: groupEvent.getBackgroundColorB(),
+      };
+
+      return backgroundColors;
+    } else {
+      console.error(
+        'Dialog was opened for an unsupported event background color'
+      );
+      return '';
+    }
+  };
+
+  state = {
+    textValue: this.getComment(),
+    textColors: this.getTextColors(),
+    backgroundColors: this.getBackgroundColors(),
+  };
+
+  render() {
+    const { event, onApply, onClose } = this.props;
+    const eventType = event.getType();
+
+    return (
+      <Dialog
+        title={<Trans>Text editor</Trans>}
```

Comment:
A bit skeptical about this, the text editor might be confused with the editor of a text object... maybe "Edit the event text"?